lib/ansible/modules/network/frr/frr_facts.py

```diff
@@ -0,0 +1,394 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2019, Ansible by Red Hat, inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+DOCUMENTATION = """
+---
+module: frr_facts
+version_added: "2.8"
+author: "Nilashish Chakraborty (@nilashishc)"
+short_description: Collect facts from remote devices running Free Range Routing (FRR).
+description:
+  - Collects a base set of device facts from a remote device that
+    is running FRR.  This module prepends all of the
+    base network fact keys with C(ansible_net_<fact>).  The facts
+    module will always collect a base set of facts from the device
+    and can enable or disable collection of additional facts.
+notes:
+  - Tested against FRR 6.0.
+options:
+  gather_subset:
+    description:
+      - When supplied, this argument restricts the facts collected
+         to a given subset.
+      - Possible values for this argument include
+         C(all), C(hardware), C(config), and C(interfaces).
+      - Specify a list of values to include a larger subset.
+      - Use a value with an initial C(!) to collect all facts except that subset.
+    required: false
+    default: '!config'
+"""
+
+EXAMPLES = """
+# Collect all facts from the device
+- frr_facts:
+    gather_subset: all
+
+# Collect only the config and default facts
+- frr_facts:
+    gather_subset:
+      - config
+
```

Comment:
It's unclear to me how to collect two sets of facts, say hardware and config.  That might be a more enlightening example here?

Reply:
Agreed. I have updated the examples.